client/state/data-layer/wpcom/sites/automated-transfer/initiate/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { initiateTransfer, requestEligibility } from 'state/automated-transfer/actions';
+import { transferStates } from 'state/automated-transfer/constants';
+import {
+	AUTOMATED_TRANSFER_INITIATE_REQUEST,
+} from 'state/action-types';
+
+/**
+ * Maps from API response to internal representation of automated transfer initialize request
+ *
+ * @param {Object} data API response data
+ * @returns {Object} Calypso eligibility information
+ */
+const fromApi = data => Object.assign(
+	{
+		id: data.transfer_id,
+	},
+	data.status === 'active' && { status: transferStates.START },
+	data.status === 'uploading' && { status: transferStates.START },
+	data.active_theme_slug && data.active_theme_slug.length && { theme: { id: data.active_theme_slug } },
+);
+
+/**
+ * Dispatches to initiate an automated transfer for a site
+ *
+ * @param {Function} dispatch action dispatcher
+ * @param {number} siteId site for which the update belongs
+ * @returns {Function} the handler function with site and dispatch partially applied
+ */
+const apiResponse = ( dispatch, siteId ) => data => dispatch( initiateTransfer( siteId, fromApi( data ) ) );
+
+/**
+ * Respond to API fetch failure
+ *
+ * @param {Object} error error from API fetch
+ */
+const apiFailure = ( dispatch, siteId ) => () => dispatch( requestEligibility( siteId ) );
+
+/**
+ * Issues an API request to initiate an automatic transfer
+ *
+ * @param {Function} dispatch action dispatcher
+ * @param {number} siteId which site to transfer
+ * @param {string} pluginSlug name of community plugin initiating transfer
+ * @param {File} themeFile zip of theme initiating transfer
+ * @returns {Promise} response promise from API fetch
+ */
+export const requestTransfer = ( { dispatch }, { siteId, pluginSlug, themeFile } ) =>
```

Comment:
Already there and already [being used](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/actions.js#L500).

I didn't know we'd stopped using `undocumented.js`. I don't mind where it lives as long as it carries on working. There are [unit tests](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/test/actions.js#L713), but these don't check that a theme zip actually gets uploaded. That has to be tested with a real call, which I'm happy to do if you want to consolidate these calls.